### ISSUE TITLE
Suggestion: File Expiration

### ISSUE THREAD

**Issue by Pinaz993** [dividebysandwich/shareboxx#2]
It may be nice, in case of limited storage space, for an opt-in option for files to expire after a configurable amount of time, after which they are cleared from the file system. Perhaps user configurable expiration times as well, with an upper limit of whatever the default is.

**Comment by dividebysandwich** [comment]
Hmmmm... not sure I like that idea. Though I can see the need for some sort of "management" of uploaded files. I was thinking about a simple "admin mode" that lets authenticated users move around and delete files, what do you think?

**Comment by Pinaz993** [comment]
Perhaps a more deliberate approach is needed, yes. Maybe the management
interface could, on a file-by-file basis, set expirations as one option?
Deleting outright is of course, preferable in most cases, I think.

I was thinking more of unattended setups, though. A management interface is
great if you can lay hands on or remote into the box, but if you're
geocaching, that might not be a viable option.

**Comment by dividebysandwich** [comment]
I'm trying to understand the use case a bit more. What is the reason why you'd want files to go away after a set time? Is it just to prevent the disk from filling up? In other words, if someone uploads something, someone connecting a month later would see it, but someone connecting after 2 months might not. If nobody uploads something for a longer time, you end up with an empty disk and nothing for anyone to download...

**Comment by Pinaz993** [comment]
Let's say you have a geocache. I seem to remember an image of a USB
plug/port set into mortar, perhaps from your video?

The cache is remote. It's not easy fo you to reach. People are enjoying the
messages and files being left.

Next thing you know, you get reports that it's broken. No one can upload.
You make the effort to go lay hands on the box, and you find out that
someone uploaded a 10-hour video of Never Gonna Give You Up, and that it
took up 90% of the disk space. You delete it. It works for a while. Then
the guy does it again. You get the picture.

Perhaps, to moderate such issues, you can have an option so that files
above a certain configurable threshold of size will expire after a certain
amount of time. Perhaps, even make that amount of time depend on how large
the file is. If someone wants to use your box as a dead-drop for large
files, they still can, but if those files are too large, they'll only
survive for a day or so, whereas an ASCII painting of a cool dragon won't
expire at all.

**Comment by dividebysandwich** [comment]
How about something like this:
Files uploaded by anonymous users can be set to expire after a set amount time (optional parameter, could default to never expire)
If you manage to get to your shareboxx and connect, you can use a password to go into "management mode" and "approve" uploaded files so they stay indefinitely. I was thinking about doing this based on folders but I think it needs to be decided per file.

There's not too much you can do against willful sabotage. Even if you set the expiration to a week, the perpetrator could be coming by once a week just to fill up whatever quota exists. If you only restrict large files, they can just fill up with lots of smaller files, etc.
But if they only come by once or twice, expiration would ensure the system fixes itself. And being able to "approve" files to keep them from expiring would make it possible to preserve "valuable" uploads.

What do you think?

**Comment by Pinaz993** [comment]
You're right about not being able to do much about willful sabotage. More
management tools are definitely welcome. And of course, it's all
configurable, so no one is forced to use it. Sounds good to me!

**Comment by dividebysandwich** [comment]
Implemented in v1.1.0/v1.1.1

<img width="435" height="733" alt="Image" src="https://github.com/user-attachments/assets/3beb9a93-ccb1-4fb5-b286-fa62fae63e0d" />

**Comment by Pinaz993** [comment]
That was fast!
